runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/ThreeStepCrypticClue.java

```diff
@@ -0,0 +1,203 @@
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import net.runelite.client.util.Text;
+
+@Getter
+public class ThreeStepCrypticClue extends ClueScroll implements TextClueScroll, ObjectClueScroll, NpcClueScroll, LocationsClueScroll
+{
+
+	private List<Map.Entry<CrypticClue, Boolean>> clueSteps;
+	private List<Integer> objectIds = new ArrayList<>();
+	private List<WorldPoint> locations = new ArrayList<>();
+	private List<String> npcs = new ArrayList<>();
+	private String text;
+
+	private ThreeStepCrypticClue(List<Map.Entry<CrypticClue, Boolean>> steps, String text)
+	{
+		this.clueSteps = steps;
+		this.text = text;
+
+		for (Map.Entry<CrypticClue, Boolean> e : clueSteps)
+		{
+			CrypticClue c = e.getKey();
+
+			objectIds.add(c.getObjectId());
+			locations.add(c.getLocation());
+			npcs.add(c.getNpc());
+		}
+	}
+
+	@Override
+	public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plugin)
+	{
+		panelComponent.setPreferredSize(new Dimension(200, 0));
+
+		for (int i = 0; i < 3; i++)
+		{
+
+			Map.Entry<CrypticClue, Boolean> e = clueSteps.get(i);
+
+			if (!e.getValue())
+			{
+				CrypticClue c = e.getKey();
+
+				panelComponent.getChildren().add(TitleComponent.builder().text("Cryptic Clue #" + (i + 1)).build());
+				panelComponent.getChildren().add(LineComponent.builder().left("Solution:").build());
+				panelComponent.getChildren().add(LineComponent.builder()
+					.left(c.getSolution())
+					.leftColor(TITLED_CONTENT_COLOR)
+					.build());
+			}
+		}
+	}
+
+	@Override
+	public void makeWorldOverlayHint(Graphics2D graphics, ClueScrollPlugin plugin)
+	{
+		for (Map.Entry<CrypticClue, Boolean> e : clueSteps)
+		{
+			if (!e.getValue())
+			{
+				e.getKey().makeWorldOverlayHint(graphics, plugin);
+			}
+		}
+	}
+
+	public void checkForParts(Client client, final ItemContainerChanged event, ItemManager itemManager)
+	{
+		if (event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
+		{
+			checkForPart(event, itemManager, TORN_CLUE_SCROLL_PART_1, 0);
+			checkForPart(event, itemManager, TORN_CLUE_SCROLL_PART_2, 1);
+			checkForPart(event, itemManager, TORN_CLUE_SCROLL_PART_3, 2);
+		}
+	}
+
+	private void checkForPart(final ItemContainerChanged event, ItemManager itemManager, int clueScrollPart, int index)
+	{
+		final Stream<Item> items = Arrays.stream(event.getItemContainer().getItems());
+
+		// If we have the part then that step is done
+		if (items.anyMatch(item -> itemManager.getItemComposition(item.getId()).getId() == clueScrollPart))
+		{
+			clueSteps.get(index).setValue(true);
+		}
+	}
+
+	@Override
+	public void reset()
+	{
+		this.locations.clear();
+	}
+
+	public static ThreeStepCrypticClue forText(String text)
+	{
+		// used for checking if a clue equals this one
+		String plainText = Text.removeTags(text
```

Comment:
This is doing the exact same thing as https://github.com/runelite/runelite/blob/f97e06d0275c6ea0d9ec8da707ce75b203e6191b/runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java#L448 so you can pass in both ``clueScrollText.getText()`` and ``text`` when you're calling forText instead of computing this twice.